game/graphics/effect.cpp

```diff
@@ -312,9 +312,15 @@ void Effect::onCollide(World& world, const VisualFx* root, const Vec3& pos, Npc*
     eff.setOrigin(other);
     eff.setActive(true);
 
-    if(npc!=nullptr)
-      npc ->runEffect(std::move(eff)); else
+    if(npc!=nullptr) {
+      npc ->runEffect(std::move(eff));
+      if(vfx->sendAssessMagic) {
+        auto oth = other==nullptr ? npc : other;
+        npc->perceptionProcess(*oth,npc,0,PERC_ASSESSMAGIC);
```

Comment:
There is another call to `PERC_ASSESSMAGIC` in code bellow.
Would it cause double perception, if both `vsm!=nullptr` and `root->emFXCollDynPerc!=nullptr` ?

Reply:
I went the lazy way and thought the modder is responisble to make sure only one is triggered. But reality is `emFXCollDynPerc` attribute doesn't exist in G1 and is only used in G2.  And in my testing `emFXCollDyn` perc couldn't be triggered in G2, so each Gothic has it's own vfx for percs.

Tested the spells in rockefellers inventory and there wasn't a single spell where `sendAssessMagic` was true for both `emFXCollDyn` and `emFXCollDynPerc`.